File: app/src/main/java/abandonedstudio/app/cookingapp/Database/DishCategoryDao.java
package abandonedstudio.app.cookingapp.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface DishCategoryDao {

    @Insert
    void insert(DishCategory dishCategory);

    @Update
    void update(DishCategory dishCategory);

    @Delete
    void delete(DishCategory dishCategory);

    @Query("SELECT * FROM dish_category ORDER BY category COLLATE NOCASE")
    LiveData<List<DishCategory>> getAllCategories();
}

File: app/src/main/java/abandonedstudio/app/cookingapp/core/data/local/room/dish_category/DishCategoryDao.kt
package abandonedstudio.app.cookingapp.core.data.local.room.dish_category

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface DishCategoryDao {

    @Insert
    suspend fun insert(dishCategory: DishCategory)

    @Update
    suspend fun update(dishCategory: DishCategory)

    @Delete
    suspend fun delete(dishCategory: DishCategory)

    @Query("SELECT * FROM dish_category ORDER BY category COLLATE NOCASE")
    fun getAllCategories(): Flow<List<DishCategory>>
}